src/traces/scattermapbox/convert.js

```diff
@@ -331,3 +361,48 @@ function blankFillFunc() { return ''; }
 function isBADNUM(lonlat) {
     return lonlat[0] === BADNUM;
 }
+
+function createClusterColors(colors, step) {
+    var idx, newColors;
+    if(Lib.isArrayOrTypedArray(colors)) {
```

Comment:
Here we should also check for `step` to be an array.
On another note. Wondering if you could merge/reuse all/any of these 3 functions: `createClusterColors`, `createClusterSizes` and `createClusterOpacities`?

Reply:
I'm pretty sure it is possible. I'll take a look at it